runtime/stdlib/contracts/test.cdc

```diff
@@ -95,6 +95,38 @@ pub contract Test {
         pub fun useConfiguration(_ configuration: Configuration) {
             self.backend.useConfiguration(configuration)
         }
+
+        /// Returns all the logs from the blockchain, up to the calling point.
+        ///
+        pub fun logs(): [String] {
+            return self.backend.logs()
+        }
+
+        /// Returns the service account of the blockchain. Can be used to sign
+        /// transactions with this account.
+        ///
+        pub fun serviceAccount(): Account {
+            return self.backend.serviceAccount()
+        }
+
+        /// Returns all events emitted from the blockchain.
+        ///
+        pub fun events(): [AnyStruct] {
+            return self.backend.events("")
+        }
+
+        /// Returns all events emitted from the blockchain,
+        /// filtered by type name.
+        ///
+        pub fun eventsOfType(_ typeName: String): [AnyStruct] {
+            return self.backend.events(typeName)
```

Comment:
That's a good idea :+1: I was worried that it will be impossible to pass a `Type` from a test script, because in integration tests, contracts are simply deployed to an address, and there is no import happening. Hence, it would not be possible to pass the concrete run-time type.

I found a workaround though:
```cadence
let type: Type = CompositeType("A.0ae53cb6e3f42a79.FlowToken.TokensDeposited")!
type == Type<FlowToken.TokensDeposited>() // => true
type.isSubtype(of: Type<FlowToken.TokensDeposited>()) // true
```

What do you think?

Reply:
That looks good to me!

Reply:
I will change the API to accept a `Type` then, instead of a `String`

Reply:
Updated here: https://github.com/onflow/cadence/pull/2505/files#diff-d3a08c1a13225bf0d1cee4f91a918d595e21c64c9fcfbcc1ac134d281dcf3a18R118-R123

For the test provider interface, I still use the type identifier, in order to match the relevant method signature from the emulator's blockchain:
- https://github.com/onflow/flow-emulator/blob/master/blockchain.go#L1077-L1078
- https://github.com/onflow/cadence-tools/pull/113/files#diff-250eb1fa6e3bb9b26ead394f26f14651772ad393772fc350e91608d608bba0adR497-R502

Reply:
Nice!